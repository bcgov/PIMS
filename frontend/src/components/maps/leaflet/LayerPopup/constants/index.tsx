import * as React from 'react';

export const MUNICIPALITY_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP';
export const PARCELS_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW/wfs?service=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW';

export const parcelLayerPopupConfig = {
  PARCEL_NAME: { label: 'Parcel Name:', display: (data: any) => data.PARCEL_NAME },
  PIN: { label: 'Parcel PIN:', display: (data: any) => data.PIN },
  PID: { label: 'Parcel PID:', display: (data: any) => data.PID },
  PLAN_NUMBER: { label: 'Plan Number:', display: (data: any) => data.PLAN_NUMBER },
  PARCEL_CLASS: { label: 'Parcel Class:', display: (data: any) => data.PARCEL_CLASS },
  MUNICIPALITY: { label: 'Municipality:', display: (data: any) => data.MUNICIPALITY },
  REGIONAL_DISTRICT: {
    label: 'Regional District:',
    display: (data: any) => data.REGIONAL_DISTRICT,
  },
  FEATURE_AREA_SQM: {
    label: 'Area:',
    display: (data: any) => (
      <>
        {data.FEATURE_AREA_SQM} m<sup>2</sup>
      </>
    ),
  },
};

export const municipalityLayerPopupConfig = {
  ADMIN_AREA_GROUP_NAME: {
    label: 'Administration Area:',
    display: (data: any) => `${data.ADMIN_AREA_GROUP_NAME} (${data.CHANGE_REQUESTED_ORG})`,
  },
};
