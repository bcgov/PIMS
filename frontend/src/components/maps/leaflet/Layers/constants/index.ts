export const MUNICIPALITY_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP';
export const PARCELS_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW/wfs?service=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW';

export const parcelLayerPopupConfig = {
  PIN: { label: 'Parcel PIN:', display: (data: any) => data.PIN },
  PID: { label: 'Parcel PID:', display: (data: any) => data.PID },
  REGIONAL_DISTRICT: {
    label: 'Regional District:',
    display: (data: any) => data.REGIONAL_DISTRICT,
  },
};

export const municipalityLayerPopupConfig = {
  ADMIN_AREA_NAME: { label: 'Area Name:', display: (data: any) => data.ADMIN_AREA_NAME },
  ADMIN_AREA_ABBREVIATION: {
    label: 'Area Abbrevation:',
    display: (data: any) => data.ADMIN_AREA_ABBREVIATION,
  },
  ADMIN_AREA_GROUP_NAME: {
    label: 'Area Group Name:',
    display: (data: any) => `${data.ADMIN_AREA_GROUP_NAME} (${data.CHANGE_REQUESTED_ORG})`,
  },
};
