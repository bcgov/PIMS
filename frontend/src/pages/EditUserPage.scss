.userInfo {
    width: 40%;
    vertical-align: middle;
    text-align: left;
    display: table;
    
}

.notes {
    padding-bottom: 30px;
}

.isDisabled {
    padding-bottom: 10px;
}

.cancelSave {
    padding-top: 15px;
    padding-bottom: 10px;
}
.navBar {
    font-weight: 700;
    text-align: left;
}

.required
{
    color: red;
    font-weight: 700;
}



