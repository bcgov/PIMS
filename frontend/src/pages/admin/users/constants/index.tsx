import * as React from 'react';
import { MUIDataTableColumnDef } from 'mui-datatables';
import { FaCheck, FaTimes } from 'react-icons/fa';
import { AccountActive } from '../interfaces/IUserRecord';
import { RowActions } from '../components/RowActions';
export const columns: MUIDataTableColumnDef[] = [
  {
    name: 'id',
    label: 'ID',
    options: {
      display: 'excluded',
      filter: false,
    },
  },
  {
    name: 'username',
    label: 'IDIR/BCeID',
    options: {
      customBodyRender: (username: string, tableMeta: any) => {
        return <a href={`/admin/user/${tableMeta.rowData[0]}`}>{username}</a>;
      },
    },
  },
  {
    name: 'firstName',
    label: 'First name',
  },
  {
    name: 'lastName',
    label: 'Last name',
  },
  {
    name: 'email',
    label: 'Email',
  },
  {
    name: 'active',
    label: 'Active',
    options: {
      customBodyRender: (active: AccountActive, tableMeta: any) => {
        return active === AccountActive.YES ? <FaCheck /> : <FaTimes />;
      },
    },
  },
  {
    name: 'agency',
    label: 'Agency',
  },
  {
    name: 'position',
    label: 'Position',
  },
  {
    name: 'role',
    label: 'Role',
  },
  {
    name: 'action',
    label: 'Actions',
    options: {
      customBodyRender: (value: any, tableMeta: any) => {
        return <RowActions userId={tableMeta.rowData[0]} active={tableMeta.rowData[5]} />;
      },
    },
  },
];
