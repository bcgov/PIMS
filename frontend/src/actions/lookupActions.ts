import * as ActionTypes from 'constants/actionTypes';

//Parcel List API action

export interface ILookupCode {
  code: string;
  name: string;
  id: string;
  isDisabled: boolean;
  isPublic?: boolean;
  type: string;
  parentId?: number;
}

export interface IStoreLookupCodeAction {
  type: typeof ActionTypes.STORE_LOOKUP_CODE_RESULTS;
  lookupCodes: ILookupCode[];
}

export const storeLookupCodesAction = (lookupCodes: ILookupCode[]) => ({
  type: ActionTypes.STORE_LOOKUP_CODE_RESULTS,
  lookupCodes: lookupCodes,
});
