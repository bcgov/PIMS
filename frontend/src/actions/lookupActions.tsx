import * as ActionTypes from 'constants/actionTypes';

//Parcel List API action

export interface ILookupCode {
  code: string;
  name: string;
  id: string;
  isDisabled: boolean;
  type: string;
}

export interface IStoreLookupCodeAction {
  type: typeof ActionTypes.STORE_LOOKUP_CODE_RESULTS;
  lookupCodes: ILookupCode[];
}

export const storeLookupCodesAction = (lookupCodes: ILookupCode[]) => ({
  type: ActionTypes.STORE_LOOKUP_CODE_RESULTS,
  lookupCodes: lookupCodes,
});
