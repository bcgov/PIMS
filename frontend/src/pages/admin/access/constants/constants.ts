import { ColDef } from 'ag-grid-community';
import { IAccessRequest } from 'interfaces';
import { AccessStatusDisplay, AccessStatusDisplayMapper } from 'constants/accessStatus';

export const columnDefinitions = (): ColDef[] => {
  let items: ColDef[] = [
    {
      headerName: 'IDIR/BCeID',
      checkboxSelection: true,
      headerCheckboxSelection: true,
      valueGetter: params => (params.data as IAccessRequest).user.username,
    },
    {
      headerName: 'First name',
      valueGetter: params => (params.data as IAccessRequest).user.firstName,
    },
    {
      headerName: 'Last name',
      valueGetter: params => (params.data as IAccessRequest).user.lastName,
    },
    {
      headerName: 'Email',
      valueGetter: params => (params.data as IAccessRequest).user.email,
      width: 250,
    },
    {
      headerName: 'Position',
      valueGetter: params => (params.data as IAccessRequest).position,
    },
    {
      headerName: 'Status',
      valueGetter: (params): AccessStatusDisplay =>
        AccessStatusDisplayMapper[(params.data as IAccessRequest).status],
      filter: false,
      sortable: false,
    },
    {
      headerName: 'Agency',
      valueGetter: params => (params.data as IAccessRequest).agencies[0].name,
      width: 250,
    },
    {
      headerName: 'Role',
      valueGetter: params => (params.data as IAccessRequest).roles[0].name,
    },
  ];

  items = items.map(column => ({
    filter: true,
    suppressMenu: true,
    cellClass: 'user-info-cell',
    sortable: true,
    minWidth: 230,
    ...column,
  }));

  return items;
};
