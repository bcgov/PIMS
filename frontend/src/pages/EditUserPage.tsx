import React, { useEffect } from 'react';
import {
  Navbar,
  Container,
  Row,
  Col,
  InputGroup,
  FormControl,
  ButtonToolbar,
  Button,
} from 'react-bootstrap';
import { Form, Input, Select, SelectOption } from '../components/common/form';
import { fetchUserDetail, getUpdateUserAction } from 'actionCreators/usersActionCreator';
import { useDispatch, useSelector } from 'react-redux';
import { RootState } from 'reducers/rootReducer';
import { IUserDetails } from 'actions/adminActions';
import _ from 'lodash';
import { Formik, ErrorMessage } from 'formik';
import { UserUpdateSchema } from 'utils/YupSchema';
import { IUserDetailParams } from 'constants/API';
import { ILookupCodeState } from 'reducers/lookupCodeReducer';
import { ILookupCode } from 'actions/lookupActions';
import * as API from 'constants/API';
import './EditUserPage.scss';
import { Label } from 'components/common/Label';

const EditUserPage = (props: IUserDetailParams) => {
  useEffect(() => {
    dispatch(fetchUserDetail({ id: props.id }));
  }, []);

  const dispatch = useDispatch();

  const lookupCodes = useSelector<RootState, ILookupCode[]>(
    state => (state.lookupCode as ILookupCodeState).lookupCodes,
  );

  const agencies = _.filter(lookupCodes, (lookupCode: ILookupCode) => {
    return lookupCode.type === API.AGENCY_CODE_SET_NAME;
  });
  const roles = _.filter(lookupCodes, (lookupCode: ILookupCode) => {
    return lookupCode.type === API.ROLE_CODE_SET_NAME;
  });

  const user = useSelector<RootState, IUserDetails>(state => state.GET_USER_DETAIL as IUserDetails);

  const getAgencyIdToPut = (name: any) => {
    let idToSend = '';
    agencies.map(val => {
      if (name === val.name) {
        idToSend = val.id;
      }
    });
    return idToSend;
  };

  const getRoleIdToPut = (name: any) => {
    let idToSend = '';
    roles.map(val => {
      if (name === val.name) {
        idToSend = val.id;
      }
    });
    return idToSend;
  };

  const mapLookupCode = (code: ILookupCode): SelectOption => ({
    label: code.name,
    value: code.id.toString(),
  });

  const selectAgencies = agencies.map(c => mapLookupCode(c));
  const selectRoles = roles.map(c => mapLookupCode(c));

  // Arrays below are used to add the role/agency from the dropdown later in code
  let agenciesToUpdate: any[];
  let rolesToUpdate: any[];

  let checkAgencies: {} | null | undefined;
  let checkRoles: {} | null | undefined;

  checkAgencies =
    user.agencies.length > 0 ? (
      <Select field="agency" placeholder={user.agencies[0].name} options={selectAgencies} />
    ) : (
      <Select field="agency" placeholder="Please Select" options={selectAgencies} />
    );

  checkRoles =
    user.roles.length > 0 ? (
      <Select field="role" placeholder={user.roles.find(x => x).name} options={[]} />
    ) : (
      <Select field="role" placeholder="Please Select" options={[]} />
    );

  const initialValues = {
    username: user.username,
    firstName: user.firstName,
    lastName: user.lastName,
    email: user.email,
    displayName: user.displayName,
    isDisabled: false,
    rowVersion: user.rowVersion,
    emailVerified: false,
    agencies: user.agencies,
    roles: user.roles,
    note: user.note,
    agency: '',
    role: '',
  };

  return (
    <div>
      <Navbar className="navBar" expand="sm" variant="light" bg="light">
        <Navbar.Brand href="#">User Information</Navbar.Brand>
      </Navbar>
      <Container fluid={true}>
        <Row className="justify-content-md-center">
          <Formik
            enableReinitialize
            initialValues={initialValues}
            validationSchema={UserUpdateSchema}
            onSubmit={(values, { setSubmitting }) => {
              if (values.agency !== '') {
                agenciesToUpdate = [{ id: Number(values.agency) }];
              } else {
                agenciesToUpdate = user.agencies;
              }

              if (values.role !== '') {
                rolesToUpdate = [{ id: values.role }];
              } else {
                rolesToUpdate = user.roles;
              }

              dispatch(
                getUpdateUserAction(
                  { id: props.id },
                  {
                    id: user.id,
                    username: user.username,
                    displayName: values.displayName,
                    firstName: values.firstName,
                    lastName: values.lastName,
                    email: values.email,
                    isDisabled: values.isDisabled,
                    rowVersion: values.rowVersion,
                    emailVerified: values.emailVerified,
                    agencies: agenciesToUpdate,
                    roles: rolesToUpdate,
                  },
                ),
              );
              setSubmitting(false);
            }}
          >
            {props => (
              <Form className="userInfo">
                <Label>IDIR/BCeID</Label>
                <Input field="username" value={props.values.username} readOnly={true} type="text" />

                <Row>
                  <Col>
                    <Label>First Name</Label>
                    <Input field="firstName" placeholder={props.values.firstName} type="text" />
                  </Col>
                  <Col>
                    <Label>Last Name</Label>
                    <Input field="lastName" placeholder={props.values.lastName} type="text" />
                  </Col>
                </Row>

                <Label>Email</Label>
                <Input field="email" placeholder={props.values.email} type="email" />

                <Label>Agency</Label>
                {checkAgencies}

                <Label>Position</Label>
                <Input
                  field="position"
                  placeholder="e.g) Director, Real Estate and Stakeholder Engagement"
                  type="text"
                />

                <Label>Role</Label>
                {checkRoles}

                <Label>Notes</Label>
                <InputGroup className="notes">
                  <FormControl
                    as="textarea"
                    placeholder="Please specify why you need access to this site."
                  />
                </InputGroup>

                <InputGroup className="isDisabled">
                  <InputGroup.Prepend>
                    <InputGroup.Text>Would you like to disable this user?</InputGroup.Text>
                  </InputGroup.Prepend>
                  <InputGroup.Append>
                    <InputGroup.Checkbox
                      onClick={() => (props.values.isDisabled = true)}
                      name="isDisabled"
                    />
                  </InputGroup.Append>
                </InputGroup>

                <Label>Status</Label>
                <Select
                  field="status"
                  placeholder="Please Select"
                  options={[
                    { value: 0, label: 'Approved' },
                    { value: 1, label: 'Denied' },
                  ]}
                />

                <Row className="justify-content-md-center">
                  <ButtonToolbar className="cancelSave">
                    <Button className="mr-5" variant="outline-dark" type="reset">
                      Cancel
                    </Button>
                    <Button className="mr-5" type="submit">
                      Save
                    </Button>
                  </ButtonToolbar>
                </Row>
              </Form>
            )}
          </Formik>
        </Row>
      </Container>
    </div>
  );
};

export default EditUserPage;
