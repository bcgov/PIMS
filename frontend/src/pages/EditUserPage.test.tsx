import EditUserPage from './EditUserPage';
import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { ILookupCode } from 'actions/lookupActions';
import * as API from 'constants/API';
import { Provider } from 'react-redux';
import * as reducerTypes from 'constants/reducerTypes';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import { render } from '@testing-library/react';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureMockStore([thunk]);
const history = createMemoryHistory();

const lCodes = {
  lookupCodes: [
    { name: 'agencyVal', id: '1', isDisabled: false, type: API.AGENCY_CODE_SET_NAME },
    { name: 'roleVal', id: '2', isDisabled: false, type: API.ROLE_CODE_SET_NAME },
  ] as ILookupCode[],
};

const selectedUser = {
  username: 'test.user',
  firstName: 'Test',
  lastName: 'User',
  email: 'test@user.com',
  isDisabled: false,
  emailVerified: false,
  agencies: [],
  roles: [{ id: '2' }],
  rowVersion: 'AAAAAAAAB9E=',
  note: 'test note',
};

const store = mockStore({
  [reducerTypes.GET_USER_DETAIL]: selectedUser,
  [reducerTypes.LOOKUP_CODE]: lCodes,
});

it('EditUserPage renders', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router history={history}>
          <EditUserPage id="TEST-ID" />,
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains role options from lookup code + please select disabled option', () => {
  const { getAllByText, getByTestId } = render(
    <Provider store={store}>
      <Router history={history}>
        <EditUserPage id="TEST-ID" />,
      </Router>
    </Provider>,
  );
  expect(getAllByText(/Roles/i));
  expect(getAllByText(/roleVal/i));
  expect(getAllByText(/agencyVal/i));
  expect(getByTestId('isDisabled').getAttribute('value')).toEqual('false');
});

describe('appropriate fields are autofilled', () => {
  it('autofills  email, username, first and last name', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router history={history}>
          <EditUserPage id="TEST-ID" />,
        </Router>
      </Provider>,
    );
    expect(getByTestId('email').getAttribute('value')).toEqual('test@user.com');
    expect(getByTestId('username').getAttribute('value')).toEqual('test.user');
    expect(getByTestId('firstName').getAttribute('value')).toEqual('Test');
    expect(getByTestId('lastName').getAttribute('value')).toEqual('User');
    expect(getByTestId('lastName').getAttribute('value')).toEqual('User');
  });
});
