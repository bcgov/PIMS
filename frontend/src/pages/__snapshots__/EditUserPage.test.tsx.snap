// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditUserPage renders 1`] = `
Array [
  <div>
    <nav
      className="navBar navbar navbar-expand-sm navbar-light bg-light"
    >
      <a
        className="navbar-brand"
        href="#"
      >
        User Information
      </a>
    </nav>
    <div
      className="container-fluid"
    >
      <div
        className="justify-content-md-center row"
      >
        <form
          className="userInfo"
          noValidate={true}
          onSubmit={[Function]}
        >
          <p
            className="label"
          >
            IDIR/BCeID
          </p>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="input-username"
              name="username"
              onChange={[Function]}
              readOnly={true}
              type="text"
              value="test.user"
            />
          </div>
          <div
            className="row"
          >
            <div
              className="col"
            >
              <p
                className="label"
              >
                First Name
              </p>
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  id="input-firstName"
                  name="firstName"
                  onChange={[Function]}
                  placeholder="Test"
                  type="text"
                  value="Test"
                />
              </div>
            </div>
            <div
              className="col"
            >
              <p
                className="label"
              >
                Last Name
              </p>
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  id="input-lastName"
                  name="lastName"
                  onChange={[Function]}
                  placeholder="User"
                  type="text"
                  value="User"
                />
              </div>
            </div>
          </div>
          <p
            className="label"
          >
            Email
          </p>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="input-email"
              name="email"
              onChange={[Function]}
              placeholder="test@user.com"
              type="email"
              value="test@user.com"
            />
          </div>
          <p
            className="label"
          >
            Agency
          </p>
          <div
            className="form-group"
          >
            <select
              className="form-control"
              id="input-agency"
              name="agency"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                Please Select
              </option>
              <option
                value="1"
              >
                agencyVal
              </option>
            </select>
          </div>
          <p
            className="label"
          >
            Position
          </p>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="input-position"
              name="position"
              onChange={[Function]}
              placeholder="e.g) Director, Real Estate and Stakeholder Engagement"
              type="text"
            />
          </div>
          <p
            className="label"
          >
            Role
          </p>
          <div
            className="form-group"
          >
            <select
              className="form-control"
              id="input-role"
              name="role"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                Please Select
              </option>
              <option
                value="1"
              >
                roleVal
              </option>
            </select>
          </div>
          <p
            className="label"
          >
            Notes
          </p>
          <div
            className="notes input-group"
          >
            <textarea
              className="form-control"
              placeholder="Please specify why you need access to this site."
            />
          </div>
          <div
            className="isDisabled input-group"
          >
            <div
              className="input-group-prepend"
            >
              <span
                className="input-group-text"
              >
                Would you like to disable this user?
              </span>
            </div>
            <div
              className="input-group-append"
            >
              <span
                className="input-group-text"
              >
                <input
                  name="isDisabled"
                  onClick={[Function]}
                  type="checkbox"
                />
              </span>
            </div>
          </div>
          <p
            className="label"
          >
            Status
          </p>
          <div
            className="form-group"
          >
            <select
              className="form-control"
              id="input-status"
              name="status"
              onChange={[Function]}
            >
              <option
                value=""
              >
                Please Select
              </option>
              <option
                value={0}
              >
                Approved
              </option>
              <option
                value={1}
              >
                Denied
              </option>
            </select>
          </div>
          <div
            className="justify-content-md-center row"
          >
            <div
              className="cancelSave btn-toolbar"
              role="toolbar"
            >
              <button
                className="mr-5 btn btn-outline-dark"
                disabled={false}
                type="reset"
              >
                Cancel
              </button>
              <button
                className="mr-5 btn btn-primary"
                disabled={false}
                type="submit"
              >
                Save
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>,
  ",",
]
`;
